src/BPGates.jl

```diff
@@ -379,6 +388,7 @@ struct CNOTPerm <: BellOp
     idx2::Int
     function CNOTPerm(s1,s2,i1,i2)
         (1 <= s1 <= 6 && 1 <= s2 <= 6) || throw(ArgumentError("The permutation index needs to be between 1 and 6."))
+        # no other checks on the BP indices? maybe that's the job of the other component
```

Comment:
do you have other checks in mind?

Reply:
Just that i1 and i2 (representing the indices of the Bell states to be manipulated) aren't checked to be inbounds, but actually I think this is a good design choice because you are decoupling the gate application from the specific Bell state you are applying it to.